pyop2/compilation.py

```diff
@@ -355,94 +401,321 @@ def get_so(self, jitmodule, extension):
             return ctypes.CDLL(soname)
 
 
-class MacCompiler(Compiler):
-    """A compiler for building a shared library on mac systems.
+class ForkingCompiler(Compiler, ABC):
+
+    compiler_versions = {}
 
-    :arg cppargs: A list of arguments to pass to the C compiler
-         (optional).
-    :arg ldargs: A list of arguments to pass to the linker (optional).
+    _cc = "mpicc"
+    _cxx = "mpicxx"
 
-    :arg cpp: Are we actually using the C++ compiler?
+    _cppflags = None
+    _ldflags = None
 
-    :kwarg comm: Optional communicator to compile the code on (only
-        rank 0 compiles code) (defaults to COMM_WORLD).
+    _cflags = None
+    _cxxflags = None
+
+    _optflags = None
+    _debugflags = None
```

Comment:
Shouldn't these be deleted in favour of `default_xxxflags` in the `Compiler` class?